subsys/zbus/zbus.c

```diff
@@ -380,10 +380,8 @@ int zbus_chan_pub(const struct zbus_channel *chan, const void *msg, k_timeout_t
 
 	_ZBUS_ASSERT(chan != NULL, "chan is required");
 	_ZBUS_ASSERT(msg != NULL, "msg is required");
-
-	if (k_is_in_isr()) {
-		timeout = K_NO_WAIT;
-	}
+	_ZBUS_ASSERT(!(k_is_in_isr() && !K_TIMEOUT_EQ(timeout, K_NO_WAIT)),
```

Comment:
I am not a big fan of ternary, but in this case, where the readability of the expression is not clear without it, I would rather suggest using something like: 
```c
_ZBUS_ASSERT(k_is_in_isr() ? K_TIMEOUT_EQ(timeout, K_NO_WAIT) : true);
```